core/templates/pages/exploration-player-page/templates/information-card-modal.controller.spec.ts

```diff
@@ -0,0 +1,124 @@
+
+describe('Information Card Modal Controller', function() {
+  var $scope = null;
+  var $uibModalInstance = null;
+  var explorationSummaryObjectFactory = null;
+
+  var expInfo = null;
+
+  beforeEach(angular.mock.module('oppia'));
+
+  beforeEach(function() {
+    explorationSummaryObjectFactory = TestBed.get(
+      ExplorationSummaryObjectFactory);
+  });
+
+  beforeEach(angular.mock.inject(function($injector, $controller) {
+    var $rootScope = $injector.get('$rootScope');
+
+    $uibModalInstance = jasmine.createSpyObj(
+      '$uibModalInstance', ['close', 'dismiss']);
+
+    expInfo = {
+      id: '0',
+      num_views: 2,
+      human_readable_contributors_summary: {
+        contributor_1: {
+          num_commits: 12
+        },
+        contributor_2: {
+          num_commits: 3
+        }
+      },
+      created_on_msec: 1581965806278.269,
+      ratings: {
+        5: 0,
+        4: 3,
+        1: 0,
+        3: 2,
+        2: 1
+      },
+      last_updated_msec: 1581965806278.183,
+      language_code: 'en',
+      category: 'Test',
+      objective: 'Dummy exploration for testing all interactions',
+      activity_type: 'exploration',
+      status: 'public',
+      thumbnail_bg_color: '#a33f40',
+      tags: [
+        'h1', 'h2', 'p', 'attrs', 'quote', 'body', 'select', 'option',
+        'form', 'section', 'input', 'img', 'div'],
+      thumbnail_icon_url: '/subjects/Lightbulb.svg',
+      community_owned: true,
+      title: 'Test of all interactions'
+    };
+
+    $scope = $rootScope.$new();
+    $controller('InformationCardModalController', {
+      $scope: $scope,
+      $uibModalInstance: $uibModalInstance,
+      ExplorationSummaryObjectFactory: explorationSummaryObjectFactory,
+      expInfo: expInfo
+    });
+  }));
+
+  it('should evaluate scope variables values correctly', function() {
+    expect($scope.averageRating).toBe(3.3333333333333335);
+    expect($scope.contributorNames).toEqual(['contributor_1', 'contributor_2']);
+    expect($scope.explorationId).toBe('0');
+    expect($scope.explorationTags).toEqual({
+      tagsToShow: [
+        'h1', 'h2', 'p', 'attrs', 'quote', 'body', 'select', 'option', 'form'],
+      tagsInTooltip: ['section', 'input', 'img', 'div']
+    });
+    expect($scope.explorationTitle).toBe('Test of all interactions');
+    expect($scope.infoCardBackgroundCss).toEqual({
+      'background-color': '#a33f40'
+    });
+    expect($scope.infoCardBackgroundImageUrl).toBe('/subjects/Lightbulb.svg');
+    expect($scope.lastUpdatedString).toBe('Feb 17');
+    expect($scope.numViews).toBe(2);
+    expect($scope.objective).toBe(
+      'Dummy exploration for testing all interactions');
+    expect($scope.explorationIsPrivate).toBe(false);
+  });
+
+  it('should get title wrapper css', function() {
+    spyOn(document, 'querySelectorAll')
+      // @ts-ignore document querySelectorAll returns more than clientWidth
+      // property according with lint settings.
```

Comment:
Not sure I understand this. What error do you get if you remove this ts-ignore?

Reply:
This error here: `Type '{ clientWidth: number; }' is missing the following properties from type 'Element': assignedSlot, attributes, classList, className, and 122 more.`, which means that the return value should have more properties than just clientWidth.